app/src/org/commcare/views/media/MediaLayout.java

```diff
@@ -342,129 +271,64 @@ private View setupImage(String imageURI, String bigImageURI) {
-                    }
-                    questionTextPane.addView(mediaPane, mediaPaneParams);
-                }
-            } else {
-                if (showImageAboveText) {
-                    mediaPaneParams.addRule(CENTER_HORIZONTAL);
-                    this.addView(mediaPane, mediaPaneParams);
-                    questionTextPaneParams.addRule(RelativeLayout.BELOW, mediaPane.getId());
-                    this.addView(questionTextPane, questionTextPaneParams);
-                } else {
-                    this.addView(questionTextPane, questionTextPaneParams);
-                    mediaPaneParams.addRule(RelativeLayout.BELOW, questionTextPane.getId());
-                    mediaPaneParams.addRule(CENTER_HORIZONTAL);
-                    this.addView(mediaPane, mediaPaneParams);
-                }
-            }
-        } else {
-            this.addView(questionTextPane, questionTextPaneParams);
+            showMissingMediaView(imageURI, "Invalid reference: " + e.getReferenceString(), false, null);
         }
     }
 
-    @SuppressWarnings("deprecation")
-    private int getScreenMinimumDimension() {
-        Display display =
-                ((WindowManager)getContext().getSystemService(Context.WINDOW_SERVICE))
-                        .getDefaultDisplay();
-
-        int width, height;
-        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.HONEYCOMB_MR2) {
-            width = display.getWidth();
-            height = display.getHeight();
-        } else {
-            Point screenDims = new Point();
-            display.getSize(screenDims);
-            width = screenDims.x;
-            height = screenDims.y;
-        }
-
-        return Math.min(width, height);
-    }
-
-    /**
-     * Creates a video view for the provided URI or an error view elaborating why the video
-     * couldn't be displayed.
-     *
-     * @param inlineVideoURI   JavaRosa Reference URI
-     * @param viewLayoutParams the layout params that will be applied to the view. Expect to be
-     *                         mutated by this method
-     */
-    private View getInlineVideoView(String inlineVideoURI, RelativeLayout.LayoutParams viewLayoutParams) {
+    private void setupInlineVideoView(String inlineVideoURI) {
         try {
             final String videoFilename = ReferenceManager.instance().DeriveReference(inlineVideoURI).getLocalURI();
-
-            int[] maxBounds = getMaxCenterViewBounds();
-
             final File videoFile = new File(videoFilename);
             if (!videoFile.exists()) {
-                return getMissingMediaView(inlineVideoURI,
+                showMissingMediaView(inlineVideoURI,
                         StringUtils.getStringRobust(getContext(), R.string.video_download_prompt),
-                        true);
+                        true,
+                        () -> {
+                            hideMissingMediaView();
```

Comment:
stylistic point that you can ignore: `hideMissingMediaView` can be called from `showMissingMediaView` to DRY the completion logic further.